src/saltext/vmware/utils/drift.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+"""
+Common functions used across modules
+"""
+import json
+
+
+def drift_report(obj1, obj2, diff_level=None):
+    """
+    Finds the drift between two objects/dictionaries
+
+    obj1
+        First dictionary
+    obj2
+        Second dictionary
+    diff_level
+        Defines on what tree level to show the drift.
+        If not specified the changes are shown in the last leaf level.
+    """
+    result_diffs = []
+    _drift_recurse_(obj1, obj2, result_diffs)
+
+    result_tree = {}
+    for tup in result_diffs:
+        _drift_tree_(tup, result_tree)
+
+    if diff_level is not None and isinstance(result_tree, dict):
+        result_subtree = {}
+        _drift_subtree_(result_tree, diff_level, result_subtree)
+        return result_subtree
+
+    return result_tree
+
+
+def _drift_tree_(diffs, result_tree):
+    branch = result_tree
+    for i in range(len(diffs)):
+        if i < (len(diffs) - 2):
+            if diffs[i] not in branch:
+                branch[diffs[i]] = {}
+            branch = branch[diffs[i]]
+        elif i == (len(diffs) - 2):
+            branch[diffs[i]] = diffs[i + 1]
+
```

Comment:
**suggestion (non-blocking)** (at least, yet) - I haven't played around with this code but `range(len(...))` is *often* the wrong way to do things.

```
for diff in diffs[:-1]:
    if diff not in branch:
        branch[diff] = {}
    branch = branch[diff]
branch[diffs[-2]] = diffs[-1]
```

Something like that might work here.

Reply:
I didn't know `range(len(...))` is wrong, I'll take note.
I would prefer to keep it as it is for now.
It's more convenient to me to think of 'i' index while I was coding (a bit old fashioned way), not how to iterate.
Otherwise it can be:
`
for diff in diffs[:-2]:
    if diff not in branch:
        branch[diff] = {}
    branch = branch[diff]
branch[diffs[-2]] = diffs[-1]
`

Reply:
done

Reply:
I'm OK with leaving it as-is right now, if you come back to this code and think about it, though, we should go ahead and refactor it :+1: